### PR TITLE
[EGO-19] Demo Config and Dummy User Data

### PR DESCRIPTION
**Pull request by joneubank**
- Created "demo" profile, when active this will:
  - Allow users to log in and be assigned admin rights
  - Not persist users in the Users table (to preserve privacy of users interacting with the demo)
- Added script to populate database with dummy users

After merging, will need to configure cron job to run the sql script to reset the user table.

See note in https://github.com/overture-stack/ego/issues/19 about running the application with the demo profile active.

### CODE REVIEW

File: src/main/java/org/overture/ego/controller/AuthController.java [hunk 1]
```diff
@@ -50,12 +49,18 @@
   @Autowired
   FacebookTokenService facebookTokenService;
 
+  // -- DEMO PROFILE CONFIGURATIONS --
+  @Value("${demo.user.save:true}")
```
Comment (ra-ver) [inline src/main/java/org/overture/ego/controller/AuthController.java]: Could we have just one property called demo

demo will be either true or false

File: src/main/java/org/overture/ego/controller/AuthController.java [hunk 2]
```diff
@@ -92,15 +97,19 @@ boolean verifyJWToken(
...
+        // Saving user controlled through config - provides mechanism to run demos without saving
+        userService.create(user);
+      }
     }
     return tokenService.generateToken(user);
   }
 
   private User createNewUser(String userName, Map authInfo) {
-    String role = "USER";
+    String role = newUserRole;
```
Comment (ra-ver) [inline src/main/java/org/overture/ego/controller/AuthController.java]: could we do this rather.. demo == true ? "ADMIN" : "USER"

Also; could you please create constant for these Strings. Sorry it wasn't done in previous versions.

File: src/main/java/org/overture/ego/controller/AuthController.java [hunk 3]
```diff
@@ -92,15 +97,19 @@ boolean verifyJWToken(
   private String generateUserToken(Map authInfo){
     val userName = authInfo.get("email").toString();
     User user = userService.getByName(userName, false);
+
     if (user == null) {
       user = createNewUser(userName,authInfo);
-      userService.create(user);
+      if (saveUsersAllowed) {
```
Comment (ra-ver) [inline src/main/java/org/overture/ego/controller/AuthController.java]: We'll need to do it for facebooktokens as well

Comment (joneubank) [reply src/main/java/org/overture/ego/controller/AuthController.java]: Controlling user creation at this point should work for facebook and google - This is in the `generateUserToken(...)` function called by both `exchangeFacebookTokenForAuth(...)` and `exchangeGoogleTokenForAuth(...)` .

**Comment by joneubank** [comment]
@ra-ver Thanks. Comments addressed.

**Comment by cheapsteak** [comment]
So psyched for this!
Could we please also add some dummy apps and groups as well?

**Comment by andricDu** [comment]
https://github.com/overture-stack/ego/issues/19

**Pull request merged**